server/sonar-server/src/main/java/org/sonar/server/qualityprofile/ws/OldRestoreAction.java

```diff
@@ -77,20 +84,20 @@ public void handle(Request request, Response response) throws Exception {
     InputStream backup = request.paramAsInputStream(PARAM_BACKUP);
     InputStreamReader reader = null;
 
-    try {
+    try (DbSession dbSession = dbClient.openSession(false)) {
       checkArgument(backup != null, "A backup file must be provided");
       reader = new InputStreamReader(backup, StandardCharsets.UTF_8);
-      BulkChangeResult result = backuper.restore(reader, null);
+      OrganizationDto defaultOrg = qProfileWsSupport.getOrganizationByKey(dbSession, null);
+      QProfileRestoreSummary result = backuper.restore(dbSession, reader, defaultOrg,null);
       writeResponse(response.newJsonWriter(), result);
     } finally {
       IOUtils.closeQuietly(reader);
       IOUtils.closeQuietly(backup);
     }
   }
 
-  private void writeResponse(JsonWriter json, BulkChangeResult result) {
-    QualityProfileDto profile = result.profile();
-    if (profile != null) {
+  private void writeResponse(JsonWriter json, QProfileRestoreSummary result) {
+    QualityProfileDto profile = result.getProfile();
       String languageKey = profile.getLanguage();
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Make this line start at column 5. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AIndentationCheck)